Wip database backend

xinbei-shen opened the pull request:
All CRUD commands and sqlite database are implemented.

xinbei-shen committed:
Implement an sqlite3 database backend

Database stores unique chat ids and recipe names when /add command is
used.

xinbei-shen committed:
Implement /add command

/add command stores all parts of a recipe (name, ingredients and
directions) in the database.

xinbei-shen committed:
Implement all CRUD functions

/add, /view, /edit and /delete commands are implemented.

However, ingredients and steps cannot be named "<< back" due to the
callback_data of the "<< back" button. Steps are also
allowed to be identical to each other, meaning that deleting one copy
will delete all other copies. Moreover, steps can only be added to the
end of the recipe for now.

ichigh0st commented:
- Should add a "add yield" choice to /edit for recipes that skipped the yield
- Implement firebase to store images and their urls in the sqlite database

xinbei-shen committed:
Allow addition of a photo to every recipe

Integration with firebase allows users to add a photo for each
recipe. The SQL database stores the url of the photo.

However, photos sent by users must first be downloaded into
the local directory before it can be uploaded to firebase. This
may cause the process to become slower.
